### DISCUSSION TITLE
Uploading an Empty Release Asset Results in `400 Bad Content-Length`

### DISCUSSION THREAD

**Discussion by calebstewart** [community/community#68261] [Apps, API and Webhooks]
### Select Topic Area

Bug

### Body

This might be an odd edge case, but internally we have a repository which when released can deploy updates to two different services. The release assets include a backup before changes, and a file containing the updates that are pushed. In summary, the release assets look like:

```
- service1-backup.json
- service1-update.json
- service2-backup.json
- service2-update.json
```

This is essentially a hold-over as we transition between two services. As a result, sometimes there are not any updates to one or the other service (although always updates to at least one). In those cases, the `serviceX-update.json` is empty, but when attempting to upload the release asset, we are receiving the following:

```
POST https://uploads.github.com/repos/org/repo/releases/release_id/assets?name=serviceX-update.json: 400 Bad Content-Length:  []
```

In our pipeline, we are using [ghr](https://github.com/tcnksm/ghr) which in-turn uses [go-github](https://github.com/google/go-github) to interact with the GitHub API. The code looks correct, and the request is prepared [here](https://github.com/google/go-github/blob/master/github/github.go#L565) for anyone curious. In this case, the `req.ContentLength` would be zero, and the body would be empty, which is a valid HTTP POST request as far as I know.

If the intention is to explicitly reject empty release assets, then I am simply requesting an update to the documentation found [here](https://docs.github.com/en/free-pro-team@latest/rest/releases/assets?apiVersion=2022-11-28#upload-a-release-asset--status-codes) so that this is a known behavior.

On the other hand, if this is unintended, then it seems like a genuine bug.

I decided not to open an issue over in `ghr` or `go-github` since it doesn't seem to be an issue with those tools or libraries, but feel free to correct me. Thanks :)

**Comment by benedictschmieder** [comment]
I have the same issue. I want to upload an empty file (because it's needed programatically) and can't.
